fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/InPredicateToEqualToRule.java

```diff
@@ -0,0 +1,55 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+
+package org.apache.doris.nereids.rules.expression.rewrite.rules;
+
+import org.apache.doris.nereids.rules.expression.rewrite.AbstractExpressionRewriteRule;
+import org.apache.doris.nereids.rules.expression.rewrite.ExpressionRewriteContext;
+import org.apache.doris.nereids.trees.expressions.EqualTo;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.InPredicate;
+
+/**
+ * Rewrite InPredicate to an EqualTo Expression, if there exists exactly one element in InPredicate.options
+ * Examples:
+ * where A in (x) ==> where A = x
+ * where A not in (x) ==> where not A = x (After ExpressionTranslator, "not A = x" will be translated to "A != x")
+ */
+public class InPredicateToEqualToRule extends AbstractExpressionRewriteRule {
+
+    public static InPredicateToEqualToRule INSTANCE = new InPredicateToEqualToRule();
+
+    @Override
+    public Expression visitInPredicate(InPredicate inPredicate, ExpressionRewriteContext context) {
+        return simplifyInnerIn(inPredicate);
+    }
+
+    private Expression simplifyInnerIn(InPredicate inPredicate) {
+        if (inPredicate.getOptions().size() != 1) {
+            return inPredicate;
+        }
+
+        Expression left = inPredicate.getCompareExpr();
+        Expression right = inPredicate.getOptions().get(0);
+
+        if (right instanceof InPredicate) {
+            right = simplifyInnerIn((InPredicate) right);
+        }
+        return new EqualTo(left, right);
```

Comment:
```suggestion
        return new EqualTo(left, right.accept(this, context));
```

Reply:
Wow! Amazing~ But I still don't understand why the left expr need to call accept() also